Userland/Libraries/LibCore/System.cpp

```diff
@@ -1302,4 +1302,19 @@ ErrorOr<void> access(StringView pathname, int mode)
 #endif
 }
 
+ErrorOr<String> readlink(StringView pathname)
+{
+    // FIXME: Try again with a larger buffer.
+    char data[PATH_MAX];
+#ifdef AK_OS_SERENITY
+    int rc = serenity_readlink(pathname.characters_without_null_termination(), pathname.length(), data, sizeof(data));
```

Comment:
I feel like the Serenity implementation of `Core::System::readlink` should just be whatever `serenity_readlink` does, i.e. we should move that implementation from LibC to here.

`serenity_readlink` seems to be a remnant of before we started moving the "nicer" alternatives to LibC syscall wrappers into LibCore and making them more "C++-like" in the process. Looking at the only other place where it's used (`Core::File`), it would seemingly also benefit greatly from `ErrorOr<>` error handling and directly passing in a `StringView`.